src/syft/ast/klass.py

```diff
@@ -91,8 +98,23 @@ def run_class_method(
             if attr_path_and_name is not None:
                 attrs[attr_name] = get_run_class_method(attr_path_and_name)
 
+        def ga(__self: Any, name: str) -> Any:
+            # we need to override the __getattribute__ of our Pointer class
+            # so that if you ever access a property on a Pointer it will not just
+            # get the wrapped run_class_method but also execute it immediately
+            # object.__getattribute__ is the way we prevent infinite recursion
+            attr = object.__getattribute__(__self, name)
+
+            # if the attr key name is in the props list from above then we know
+            # we should execute it immediately and return the result
+            if name in props:
```

Comment:
where is "props" coming from? Surely this should be on __self?

Reply:
I fear that using a global-ish variable here might cause some issues where multiple pointers (or in this case pointer types) end up accidentally sharing the same list of properties.

Reply:
Yes you are right this looks a bit sketchy and should be tested better, but to be on the safe side ill move it to be implicitly stored on the new class.

Reply:
Good call!